gprofiler/profilers/perf.py

```diff
@@ -204,9 +222,61 @@ def snapshot(self) -> ProcessToProfileData:
 
         return {
             # TODO generate appids for non runtime-profiler processes here
-            k: ProfileData(v, None, None)
+            k: ProfileData(v, None, self._get_metadata(k))
             for k, v in merge.merge_global_perfs(
                 self._perf_fp.wait_and_script() if self._perf_fp is not None else None,
                 self._perf_dwarf.wait_and_script() if self._perf_dwarf is not None else None,
             ).items()
         }
+
+
+class PerfMetadata(ApplicationMetadata):
+    def relevant_for_process(self, process: Process) -> bool:
+        return False
+
+
+class GolangPerfMetadata(PerfMetadata):
+    def relevant_for_process(self, process: Process) -> bool:
+        version = self._get_golang_version(process)
+        return version is not None
+
+    @functools.lru_cache(maxsize=128)
```

Comment:
Can increase over 128 (128 processes is less than a typical system has). Use 4096